src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -470,11 +481,29 @@ public boolean isEnqueued() {
     }
 
     /**
-     * Clears this reference object and adds it to the queue with which
-     * it is registered, if any.
+     * Clears this reference object, then attempts to add it to the queue with
+     * which it is registered, if any.
      *
-     * <p> This method is invoked only by Java code; when the garbage collector
+     * <p>If this reference was already enqueued (by the garbage collector, or a
+     * previous call to {@code enqueue}), this method is <b><i>not successful</i></b>,
+     * and returns false.
+     *
+     * <p>Memory consistency effects: Actions in a thread prior to calling
+     * {@code enqueue} <b><i>successfully</i></b>
+     * <a href="{@docRoot}/java.base/java/util/concurrent/package-summary.html#MemoryVisibility"><i>happen-before</i></a>
+     * the reference is removed from the queue by {@link ReferenceQueue#poll}
+     * or {@link ReferenceQueue#remove}.
+     *
+     * <p>This method is invoked only by Java code; when the garbage collector
```

Comment:
Perhaps something along the lines of:

```suggestion
     * <p>This method is only invoked explicitly; when the garbage collector
```

Reply:
I think the important part to convey is that the GC won't call this method. So how about _just_:
```
<p>When the garbage collector enqueues references it does so directly, without invoking this method.
```
?